README.md

```diff
@@ -0,0 +1,134 @@
+# SPQLIOS
+
+## Description
+
+The SPQLIOS library aims at providing capabilities for computing the Fast Fourier Transform (FTT/NTT) in the anticyclic ring $\mathbb{Z}[X]/(X^N +1)$ for N a power of 2. It uses AVX, AVX2 and FMA assembly vectorization instructions. This library was inspired by the [Project Nayuki](https://www.nayuki.io/page/fast-fourier-transform-in-x86-assembly) and was originally provided as part of the [TFHE: Fast Fully Homomorphic Encryption Library over the Torus](https://github.com/tfhe/tfhe) library. It is now provided as an independent library with extended features and optimizations. 
+
+The library provides the following functionalities:
+* Implementation of the FFT in the complex realm with a 64-bit representation of complex coefficients.
+* Fast multiplication of polynomials in the ring Z[X]/(X^N+1) or R[X]/(X^N+1) with up to 52 bits of precision.
+
+
+### Optimisations
+
+The library auto-detects support for AVX, AVX2, FMA, AVX512F instructions at runtime and automatically switches to the 
+most efficient implementation, without requiring to be compiled on the spot on the target machine. This allows to distribute 
+not only source but also binary distributions of the library via standard package managers (e.g. apt-install) 
+
+
+## Dependencies 
+
+Spqlios-library is a C library that can be compiled with a standard C compiler, and depends only on libc and libm. The API 
+interface can be used in a regular C code, and any other language via classical foreign APIs. 
+
+The unittests and integration tests are in an optional part of the code, and are written in C++.  These tests rely on 
+[```benchmark```](https://github.com/google/benchmark), and ```gtest``` libraries, and therefore require a C++17 compiler.
+
+Currently, the project has been tested with the gcc,g++ >= 11.3.0 compiler under Linux (x86_64). In the future, we plan to
+extend the compatibility to other compilers, platforms and operating systems. 
+
+
+## Installation
+
+To build the library with the default options, create a ```build``` folder in the top level directory and run ```make``` from inside it. This assumes that the standard tool cmake is already installed on the system, and an up-to-date c++ compiler (i.e. g++ >=11.3.0) as well.
```

Comment:
The library uses a classical `cmake` build mechanism: use cmake to create a ```build``` folder in the top level directory and run ```make``` from inside it.